<launch>
    <arg name="pep_input_file" default="$(find pep_controller)src/pep_input/obstacle_avoid.pep"/>
    <arg name="loop_rate_hz" default="10"/>

    <group ns="constants">
        <!--P object name (name) and topic name (value)-->
        <param name="cruiseSpeedLeft " value="5" />
        <param name="wL0" value="10" />
        <param name="wL1" value="8" />
        <param name="wL2" value="4" />
        <param name="wL5" value="-4" />
        <param name="wL6" value="-8" />
        <param name="wL7" value="-10" />

        <param name="cruiseSpeedRight" value="5" />
        <param name="wR0" value="-10" />
        <param name="wR1" value="-8" />
        <param name="wR2" value="-4" />
        <param name="wR5" value="4" />
        <param name="wR6" value="8" />
        <param name="wR7" value="10" />
    </group>

    <group ns="input_dev">
        <!--P object name (name) and topic name (value)-->
        <group ns="range">
            <param name="prox0" value="proximity0" />
            <param name="prox0_" value="proximity0" />

            <param name="prox1" value="proximity1" />
            <param name="prox1_" value="proximity1" />

            <param name="prox2" value="proximity2" />
            <param name="prox2_" value="proximity2" />

            <param name="prox5" value="proximity5" />
            <param name="prox5_" value="proximity5" />

            <param name="prox6" value="proximity6" />
            <param name="prox6_" value="proximity6" />

            <param name="prox7" value="proximity7" />
            <param name="prox7_" value="proximity7" />
        </group>
    </group>

    <group ns="output_dev">
        <!--cmd_vel is a default output topic that is remapped later on-->
        <!--give the controller a hint on the cmd_vel values used by the robot
        diff_drive means linear.x and angular.z are used-->
        <param name="robot_type" value="diff_drive" />
        <!--P object to motor speed mapping-->
        <group ns="cmd_vel_diff_drive">
            <param name="lw" value="leftMotorSpeed" />
            <param name="rw" value="rightMotorSpeed" />
        </group>
    </group>

	<node pkg="pep_controller" type="pep_controller.py" name="pep_control" output="screen">
        <param name="pepInputFile" value="$(arg pep_input_file)"/>
        <param name="loopRateHz" value="$(arg loop_rate_hz)"/>

        <!--remap the cmd_vel output topic to that expected by epuck_driver-->
        <remap from="cmd_vel" to="mobile_base/cmd_vel"/>
	</node>

	<!--<node pkg="rviz" type="rviz" name="rviz" />-->

</launch>
